frontend/components/DocumentPreviewModal.tsx

```
@@ -607,15 +683,30 @@ const DocumentPreviewModal: React.FC<DocumentPreviewModalProps> = ({
+          />
+        );
+      }
+
+      // For authenticated/private blobs, render DOCX/XLSX client-side.
+      if (['DOCX', 'XLSX'].includes(fileExtension) && officeHtml !== null) {
+        return (
+          <div className="w-full h-full overflow-auto bg-white p-6">
+            <div
+              className="max-w-none"
+              // Sanitized via DOMPurify
+              dangerouslySetInnerHTML={{ __html: officeHtml }}
```

Biome (2.3.14)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)